src/components/logic/AddSeparatorsLogic.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div
+    @click="updateSeparatorsInShelf"
+    class="shelf"
+    :style="{ backgroundColor: shelfBackgroundColor }"
+  >
+    <separator-1
+      v-if="computedAmountOfSeparators === 1"
+      :spaceWidth="spaceWidth"
+    ></separator-1>
+    <separator-2
+      v-if="computedAmountOfSeparators === 2"
+      :spaceWidth="spaceWidth"
+    ></separator-2>
+    <separator-3
+      v-if="computedAmountOfSeparators === 3"
+      :spaceWidth="spaceWidth"
+    ></separator-3>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import Separator1 from '../furniture/separators/Separator1.vue';
+import Separator2 from '../furniture/separators/Separator2.vue';
+import Separator3 from '../furniture/separators/Separator3.vue';
+export default {
+  components: { Separator1, Separator2, Separator3 },
+  props: ['myShelf'],
+  emits: ['updating-separators'],
+  data() {
+    return {};
+  },
+  computed: {
+    ...mapGetters(['shelfs']),
+    identifiedShelf() {
+      return this.shelfs.find((shelf) => shelf.id === this.myShelf.id);
+    },
+    computedAmountOfSeparators() {
+      return this.identifiedShelf.insideSpaces.amountOfSeparators;
+    },
+    shelfBackgroundColor() {
+      return this.identifiedShelf.insideSpaces.isUpdating === true
+        ? 'rgba(109, 206, 128, 0.2)'
+        : '';
+    },
+    widthVariationRelatedToTypeOfSeparator() {
+      const objectTypeSeparators = {
+        centered: 0,
+        left: -12 / this.computedAmountOfSeparators,
+        right: 10 / this.computedAmountOfSeparators,
+      };
+      return objectTypeSeparators[
+        this.identifiedShelf.insideSpaces.typeOfSeparators
+      ];
+    },
+    spaceWidth() {
```

Comment:
Why is all that code super similar to the other component `src/components/forms/AddSeparatorForm.vue`

That's not normal, it seems super duplicated